src/config.c

```diff
@@ -775,15 +780,18 @@ void configSetCommand(client *c) {
         if (!apply_fns[i](&errstr)) {
             serverLog(LL_WARNING, "Failed applying new configuration. Possibly related to new %s setting. Restoring previous settings.", set_configs[config_map_fns[i]]->name);
             restoreBackupConfig(set_configs, old_values, config_count, apply_fns);
+            err_arg_name = set_configs[config_map_fns[i]]->name;
             goto err;
         }
     }
     addReply(c,shared.ok);
     goto end;
 
 err:
-    if (errstr) {
-        addReplyErrorFormat(c,"Config set failed - %s", errstr);
+    if (invalid_arg_name) {
+        addReplyErrorFormat(c,"Unknown option or number of arguments for CONFIG SET - '%s'", invalid_arg_name);
+    } else if (errstr) {
+        addReplyErrorFormat(c,"CONFIG SET failed (possibly related to argument '%s') - %s", err_arg_name, errstr);
     } else {
         addReplyError(c,"Invalid arguments");
```

Comment:
Why don't we also print out the information about the bad argument here? (I think this line can only be hit when "performInterfaceSet" but doesn't provide a reason

Reply:
I agree, this should probably be similar to the print to the one above, just without an explicit `errstr`.

Reply:
> I agree, this should probably be similar to the print to the one above, just without an explicit `errstr`.

Could you please provide me one case to print the last error message: Invalid arguments so that I could add one test case.
Thanks a lot.

Reply:
I couldn't find any place where we leave `errstr` NULL on error but I think it's good to have this fallback `else` just in case someone does write such a config in the future. 
The code is written under the assumption that `err_arg_name` is always set when `invalid_arg_name` is NULL. So if that's **really** true you can have the fallback else like this:
```c
    } else {
        addReplyError(c,"CONFIG SET failed (possibly related to argument '%s')", err_arg_name);
    }
```

Reply:
Sure, thanks @yoav-steinberg